### ISSUE TITLE
Add an Icon for Visual Editor in the dropdown menu

### ISSUE THREAD

**Issue by b01xy** [thingles/foreground#224]
as Visual Editor is working fine since patch 7 :)
btw nice skin, thks
b

**Comment by paladox** [comment]
Do you have any suggestions on what the icon can be. Should it be the same as the edit buttons icon or different.

**Comment by Hutchy68** [comment]
I'll have to look at font awesome 4.5.  We skipped two releases so I think there's about 40 more icons.

**Comment by Hutchy68** [comment]
Has anyone tested to see if Visual editor is working for version Foreground 1.2 in mw 1.26? Might be  my install of 1.26.2 but Visual editor is working on vector just fine. When I swap to Foreground it  doesn't even give the edit link, just edit source link.  What's even odder is if I commented out VE in localsettings and reload the page, edit (ca-ve-edit) and edit source are there in dropdown. Reload the page again and I'm back to just the plain edit link indicating VE is disabled in dropdown.

**Comment by paladox** [comment]
Did you remember to add foreground to the list of supported skins in visual editor.

Ive tested using foreground 2.0.

You need to do something like

$wgVisualEditorSupportedSkins = array(
	'vector',
	'apex',
	'monobook',
	'minerva',
	'blueprint',
	'foreground'
);

**Comment by paladox** [comment]
Also this needs to be enabled 

// Enable by default for everybody
$wgDefaultUserOptions['visualeditor-enable'] = 1;

// Don't allow users to disable it
$wgHiddenPrefs[] = 'visualeditor-enable';

**Comment by Hutchy68** [comment]
Thanks

$wgVisualEditorSupportedSkins was buried at the bottom of my 1.23 localsettings file. Working now. 

Looks like we need just something to attach to ca-ve-edit

**Comment by paladox** [comment]
The edit button and create button show for me for visual editor. I haven't set up parsoid because my host wont allow that. But it says visual edit will work without it only if you create a page.

Could I ask how your doing it. I would like to find out why it wont work. Because it wont work for vector either.

**Comment by Hutchy68** [comment]
I didn't install parsoid either because I'm using LTS 13.10 Ubuntu not a 14+, but I can pull up the Visual Editor on edit or create. Main Page it is buggy and won't load, but other main NS pages and the User NS it loads up just fine.

Added this after the required_once

$wgVirtualRestConfig['modules']['parsoid'] = array(
  // URL to the Parsoid instance
  // Use port 8142 if you use the Debian package
 'url' => 'http://localhost:8000',
  // Parsoid "domain", see below (optional)
 'domain' => 'localhost',
  // Parsoid "prefix", see below (optional)
  'prefix' => 'localhost'
);

which I think fools it into a parsoid is available

![ve](https://cloud.githubusercontent.com/assets/3527469/11981422/608f8f84-a970-11e5-879d-f02e93af36b9.PNG)

Some of the vertical dividers are off, but it does work.

How about a plan pencil without the sheet

http://fortawesome.github.io/Font-Awesome/icon/pencil/

**Comment by Hutchy68** [comment]
![ve-drop](https://cloud.githubusercontent.com/assets/3527469/11981467/f3787d9c-a970-11e5-9487-692f1e8786bf.PNG)

**Comment by paladox** [comment]
Yes please. Thanks that would be greate.

**Comment by paladox** [comment]
The reason visual editor is like that is because of this css
.column, .columns {
	margin-left
	margin-right
}

Not sure what the css propertys are but if you set those to 0 px then that works but I doint know how to override it since important is ignored. I also doint know how to do it for only when visual editor is used.

**Comment by paladox** [comment]
I also not sure about the icons heights.

**Comment by Hutchy68** [comment]
I'll take a look.

**Comment by Hutchy68** [comment]
It's the height containers of the ui tools, fix is pretty easy and we probably want to downsize the font-size anyway. I'll be adding them.

![ve-tweaks](https://cloud.githubusercontent.com/assets/3527469/11985867/3c3a8ef0-a997-11e5-85b9-3f1f6fa71aae.PNG)

```
/*Fixes Visual Editor */
.oo-ui-widget {
    height: 45px;
}

.oo-ui-toolbar-tools {
    font-size: 90%;
}

.oo-ui-toolbar-actions > .oo-ui-buttonElement.oo-ui-labelElement > .oo-ui-buttonElement-button > .oo-ui-labelElement-label {
    line-height: 2.75em !important;
}
```

**Comment by paladox** [comment]
Ok thanks. Also action menu won't work unless the visual editor bar is down a bit.

**Comment by Hutchy68** [comment]
can't move down #content it affects when VE is not activated. It's the z-index of the button itself.

**Comment by paladox** [comment]
Oh ok. Could we ajust the z-index please. also for the echo icon too please.

Also we would need to add icon for edit and edit section also for create and create section since when visual editor is enabled it changes all the code for those. Since edit becomes the button for visualeditor and edit section because the edit button for the inbuilt editor or wikieditor.

**Comment by Hutchy68** [comment]
Z-index fixed.  For anyone brave I just pushed more fixes to PR #272.  The only thing that I have an enabled yet is the native tabs feature of foundation, otherwise it should be ready for testing.

**Comment by Hutchy68** [comment]
done, see #272
